### PR TITLE
RELATED: RAIL-3948 checkbox component

### PR DESCRIPTION
**Pull request by plavemar**
- add property to set up the size of the checkbox label.
- adjust checkbox title rendering

Added `isSmallLabel` property to be able to set up the label size within the Checkbox component. This flag defaults to true to align with original behaviour of the component. If the flag is set to `false`, size indication css classes are removed.

Checkbox title was modified to get rendered only if the `title` property is set up.

JIRA: RAIL-3948

---

Supported PR commands:

| Command                  | Description            |
| ------------------------ | ---------------------- |
| `ok to test`             | Re-run standard checks |
| `extended test`          | BackstopJS tests       |
| `extended check sonar`   | SonarQube tests        |
| `extended check cypress` | Cypress E2E tests      |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `check-extended-cypress` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/6318/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/8115/ : SUCCESS in 21m 43s

File: libs/sdk-ui-kit/src/Form/Checkbox.tsx [hunk 1]
```diff
@@ -35,11 +38,15 @@ export class Checkbox extends React.PureComponent<CheckboxProps> {
     };
 
     render(): React.ReactNode {
-        const { disabled, name, text, title, value } = this.props;
+        const { disabled, name, text, title, value, isSmallLabel } = this.props;
+
+        const labelClasses = cx("input-label-text", {
+            "gd-label-small gd-checkbox-label-small": isSmallLabel,
```
Comment (no23reason) [inline libs/sdk-ui-kit/src/Form/Checkbox.tsx:44]: I guess if `!isSmallLabel` then we should use `gd-label gd-checkbox-label`?

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/6320/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/8118/ : FAILURE in 3m 29s

**Review by no23reason — APPROVED** [review]
LGTM

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR2080

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/1824/ : SUCCESS in 1m 10s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/2105/ : SUCCESS in 20s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/906/ : SUCCESS in 19m 43s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/8120/ : SUCCESS in 23m 01s

**Pull request merged**